pkg/restore/restore.go

```diff
@@ -255,18 +258,18 @@ func (kr *kubernetesRestorer) RestoreWithResolvers(
 		}
 	}
 
-	resourceRestoreHooks, err := hook.GetRestoreHooksFromSpec(&req.Restore.Spec.Hooks)
-	if err != nil {
-		return results.Result{}, results.Result{Velero: []string{err.Error()}}
-	}
-	hooksCtx, hooksCancelFunc := go_context.WithCancel(go_context.Background())
 	waitExecHookHandler := &hook.DefaultWaitExecHookHandler{
 		PodCommandExecutor: kr.podCommandExecutor,
 		ListWatchFactory: &hook.DefaultListWatchFactory{
 			PodsGetter: kr.podGetter,
 		},
 	}
 
+	hooksWaitExecutor, err := newHooksWaitExecutor(req.Restore, waitExecHookHandler)
```

Comment:
Would it be slightly better if the `waitExecHookHandler` is created in ` newHooksWaitExecutor`?

Reply:
I prefer to leave it as-is. 

waitExecHookHandler` is an interface and likely to be implemented in multiple ways as needed. So I think it might be better to let the outside code determine the specific type and  `hooksWaitExecutor` only reference the interface.

Reply:
ok